Scripts/RemediationScripts/Remediate-SetSQLServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,886 @@
+
+        [Switch]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies no back up will be taken by the script before remediation")]
+        $SkipBackup,
+
+
+        [String]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies the path to the file to be used as input for the remediation")]
+        $FilePath,
+
+        [String]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies the path to the file to be used as input for the remediation when AutoRemediation switch is used")]
+        $Path,
+
+        [Switch]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies script is run as a subroutine of AutoRemediation Script")]
+        $AutoRemediation,
+
+        [String]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies the time of creation of file to be used for logging remediation details when AutoRemediation switch is used")]
+        $TimeStamp
+    )
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 1 of 4] Prepare to set required TLS version for SQL Servers in Subscription: [$($SubscriptionId)]"
+    if ($PerformPreReqCheck)
+    {
+        try
+        {
+            Write-Host "Setting up prerequisites..."
+            Setup-Prerequisites
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting up prerequisites. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+    }
+
+    # Connect to Azure account
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context))
+    {
+        Write-Host "No active Azure login session found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+        break
+    }
+
+   
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    
+    if(-not($AutoRemediation))
+    {
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "Subscription Name: $($context.Subscription.Name)"
+        Write-Host "Subscription ID: $($context.Subscription.SubscriptionId)"
+        Write-Host "Account Name: $($context.Account.Id)"
+        Write-Host "Account Type: $($context.Account.Type)"
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    Write-Host "To Set minimal TLS version for SQL Servers in a Subscription, Contributor or higher privileges on the SQL Servers are required." -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 4] Fetch all SQL Servers"
+
+    $sqlServerResources = @()
+    $requiredMinTLSVersion = 1.2
+
+
+    # To keep track of remediated and skipped resources
+    $logRemediatedResources = @()
+    $logSkippedResources=@()
+
+    $controlIds = "Azure_SQLDatabase_DP_Use_Secure_TLS_Version_Trial"
+
+     if($AutoRemediation)
+    {
+        if(-not (Test-Path -Path $Path))
+        {
```

Comment:
correct the indentation of the code block

Reply:
corrected indentation